C/services/common/service_security.cpp

```diff
@@ -0,0 +1,308 @@
+ * @return		True is the resource acces has been granted, false otherwise
+ */
+bool ServiceAuthHandler::verifyService(string& sName, string &sType)
+{
+	// Check m_security category item ACL value service array
+	Document doc;
+	Value arrayService;
+
+	// Parse security config with lock
+	unique_lock<mutex> cfgLock(m_mtx_config);
+
+	doc.Parse(this->m_security.getValue("ACL").c_str());
+
+	cfgLock.unlock();
+
+	if (doc["service"].IsArray())
+	{
+		arrayService = doc["service"];
+	}
+
+	if (arrayService.Size() == 0)
+	{
+		return true;
+	}
+
+	if (arrayService.Size() > 0)
+	{
+		bool serviceMatched = false;
+		bool typeMatched = false;
+
+		for (Value::ConstValueIterator it = arrayService.Begin();
+		     it != arrayService.End();
+		     ++it)
+
+		{
+			if (it->IsObject())
+			{
+				if (it->HasMember("name") &&
+				    (*it)["name"].IsString())
+				{
+					string name = (*it)["name"].GetString();
+					if (name == sName)
+					{
+						serviceMatched = true;
+						break;
+					}
+				}
+
+				if (it->HasMember("type") &&
+				    (*it)["type"].IsString())
+				{
+					string type = (*it)["type"].GetString();
+					if (type == sType)
+					{
+						typeMatched = true;
+						break;
+					}
+				}
+			}
+		}
+		return serviceMatched == true || typeMatched == true;
+	}
+
+	return false;
+}
+
+/**
+ * Authentication Middleware for PUT methods
+ *
+ * Routine first check whether the service is configured with authentication
+ *
+ * Access bearer token is then verified against FogLAMP core API endpoint
+ * JWT token claims are passed to verifyURL and verifyService routines
+ *
+ * If access is granted the input funcPUT funcion is called
+ * otherwise error response is sent to the client
+ *
+ * @param response	The HTTP Response to send
+ * @param request	The HTTP Request
+ * @param funcPUT	The function to call in case of access granted
+ */
+void ServiceAuthHandler::AuthenticationMiddlewarePUT(shared_ptr<HttpServer::Response> response,
+				shared_ptr<HttpServer::Request> request,
+        			std::function<void(
+        		        shared_ptr<HttpServer::Response>,
+        		        shared_ptr<HttpServer::Request>)> funcPUT)
+{
+	bool acl_set = this->getAuthenticatedCaller();
+	if (acl_set)
+	{
+		Logger::getLogger()->debug("This service %s has AuthenticatedCaller flag set %d",
+			this->getName().c_str(),
+			acl_set);
+		map<string, string> tokenClaims;
+		bool ret = m_mgtClient->verifyAccessBearerToken(request, tokenClaims);
+		if (!ret)
+		{
+			string responsePayload = QUOTE({ "error" : "invalid service bearer token"});
+			Logger::getLogger()->error("invalid service bearer token");
+			this->respond(response,
+					SimpleWeb::StatusCode::client_error_bad_request,
+					responsePayload);
+
+			return;
+		}
+
+		// TODO get token claims from FogLAMP issued bearer token
+		// Fill tokenClaims with examples
+		tokenClaims["aud"] = "Fledge";
+		tokenClaims["sub"] = "DISP1";		// Service Name
```

Comment:
Really! A fixed string. This in in services/common so I would expect it to be used by many services.